src/test/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManagerTest.java

```diff
@@ -0,0 +1,113 @@
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.upgrade.utils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests the {@link F47ToF5UpgradeManagerTest}
+ *
+ * @author dbernstein
+ */
+
+public class F47ToF5UpgradeManagerTest {
+
+    static final String TARGET_DIR = System.getProperty("project.build.directory");
+
+    private File createTempDir() {
+        final File tmpDir = new File(System.getProperty("java.io.tmpdir"), System.currentTimeMillis() + "");
+        tmpDir.mkdirs();
+        return tmpDir;
+    }
+
+    @Test
+    public void testUpgrade() throws Exception {
+        //prepare
+        final File tmpDir = createTempDir();
+        final File input = new File(TARGET_DIR + "/test-classes/4.7.5-export");
+        final File output = new File(tmpDir, "output");
+        output.mkdir();
+
+        final var config = new Config();
+        config.setSourceVersion(FedoraVersion.V_4_7_5);
+        config.setTargetVersion(FedoraVersion.V_5);
+        config.setInputDir(input);
+        config.setOutputDir(output);
+        //run
+        UpgradeManager upgradeManager = UpgradeManagerFactory.create(config);
+        upgradeManager.start();
+        //ensure all expected files exist
+        final String[] expectedFiles = new String[]{"rest.ttl",
+                "rest.ttl.headers",
+                "rest/external1",
+                "rest/external1/fcr%3Ametadata.ttl",
+                "rest/container1.ttl.headers",
+                "rest/container1/testbinary.binary",
+                "rest/container1/testbinary/fcr%3Ametadata.ttl",
+                "rest/container1/testbinary.binary.headers",
+                "rest/container1.ttl",
+                "rest/external1.external.headers",
+                "rest/external1.external"};
+
+        for (String f : expectedFiles) {
+            assertTrue(f + " does not exist as expected", getFile(output, f).exists());
+        }
+
+        //ensure external content has been transformed properly
+        try (FileInputStream fis = new FileInputStream(getFile(output, "rest/external1/fcr%3Ametadata.ttl"))) {
```

Comment:
This can be replaced with `final String externalContent = FileUtils.readFileToString(...)`